globus_automate_client/cli/actions.py

```diff
@@ -14,6 +16,45 @@
 app = typer.Typer(short_help="Manage Globus Automate Actions")
 
 
+class ActionInputFormat(str, Enum):
+    json = "json"
+    yaml = "yaml"
+
+
+class ActionOutputFormat(str, Enum):
+    json = "json"
+    yaml = "yaml"
+
+
+# Constant to reduce if statements making the selection
+# used in calls to format_and_echo
+DUMPERS = {
+    ActionOutputFormat.json: json.dumps,
+    ActionOutputFormat.yaml: yaml.dump,
+}
+
+
+def _process_action_body(
+    body: str, input_format: ActionInputFormat
+) -> Mapping[str, Any]:
+    """
+    Turn input strings into dicts per input format type (ActionInputFormat)
+    """
+    action_dict = None
+    if input_format is ActionInputFormat.json:
+        try:
+            action_dict = json.loads(body)
+        except json.JSONDecodeError as e:
+            raise typer.BadParameter(f"Invalid JSON: {e}")
+    elif input_format is ActionInputFormat.yaml:
+        try:
+            action_dict = yaml.safe_load(body)
+        except Exception as e:
```

Comment:
Is this the most precise Exception type that can be specified for `yaml.safe_load`?

Also, it seems like the main complication of using the same `DUMPERS` type definition for a "LOADER" usage is the possible exception types, right? One could do LOADERS where the value on the dict is the loader function and a tuple of exception types if one were so inclined. 

The LOADER bit is pretty hypothetical, doesn't really matter to me, but if we can be more specific on the exception type that seems good. If that's just what yaml.safe_load returns, just say so and I'm good here too.

Reply:
That was an oversight, I'll update it to use the proper YAML exception yaml.YAMLError. Good catch, thanks!